src/tools/compiletest/src/header.rs

```diff
@@ -189,6 +191,7 @@ mod directives {
     pub const STDERR_PER_BITWIDTH: &'static str = "stderr-per-bitwidth";
     pub const INCREMENTAL: &'static str = "incremental";
     pub const KNOWN_BUG: &'static str = "known-bug";
+    pub const MIR_UNIT_TEST: &'static str = "unit-test";
```

Comment:
Yeah, this is valid. The one thing that I'm keeping in the back of my mind as motivation for this is the potential to turn on some canonicalization passes by default in the future. At that point we will certainly want this header, since we don't want tests getting out of sync.

Besides that, I don't have an opinion either way. It totally might make sense to delay this until we have a need to turn on passes by default.